include/libcyphal/transport/multiplexer.hpp

```diff
@@ -0,0 +1,33 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_MULTIPLEXER_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_MULTIPLEXER_HPP_INCLUDED
+
+namespace libcyphal
+{
+namespace transport
+{
+
+class IMultiplexer
+{
+public:
+    IMultiplexer(IMultiplexer&&)                 = delete;
+    IMultiplexer(const IMultiplexer&)            = delete;
+    IMultiplexer& operator=(IMultiplexer&&)      = delete;
+    IMultiplexer& operator=(const IMultiplexer&) = delete;
+    virtual ~IMultiplexer()                      = default;
```

Comment:
Same here. Interface destructors should always be protected/non-virtual

Reply:
I get it for external entity interfaces (like media and multiplexer), but for our own entities we implement (and return them via `makeXxx` as unique ptrs) it still blurry which pattern to use.

Reply:
Internal implementations would not be destructed via the interface though? They would use their complete internal types.